src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SocketConnection.cs

```diff
@@ -496,37 +532,77 @@ private void ThrowIfClosed()
+        {
+            return _socket;
+        }
+
+        public IAsyncResult BeginValidate(Uri uri, AsyncCallback callback, object state)
+        {
+            return new CompletedAsyncResult<bool>(true, callback, state);
+        }
+
+        public bool EndValidate(IAsyncResult result)
+        {
+            return CompletedAsyncResult<bool>.End(result);
         }
 
-        private Exception ConvertSendException(SocketException socketException, TimeSpan remainingTime)
+        private Exception ConvertSendException(SocketException socketException, TimeSpan remainingTime, TimeSpan timeout)
         {
-            return ConvertTransferException(socketException, _sendTimeout, socketException,
-                _aborted, _timeoutErrorString, _timeoutErrorTransferOperation, this, remainingTime);
+            return ConvertTransferException(socketException, timeout, socketException,
+                TransferOperation.Write, _aborted, _timeoutErrorString, _timeoutErrorTransferOperation, this, remainingTime);
         }
 
-        private Exception ConvertReceiveException(SocketException socketException, TimeSpan remainingTime)
+        private Exception ConvertReceiveException(SocketException socketException, TimeSpan remainingTime, TimeSpan timeout)
         {
-            return ConvertTransferException(socketException, _receiveTimeout, socketException,
-                _aborted, _timeoutErrorString, _timeoutErrorTransferOperation, this, remainingTime);
+            return ConvertTransferException(socketException, timeout, socketException,
+                TransferOperation.Read, _aborted, _timeoutErrorString, _timeoutErrorTransferOperation, this, remainingTime);
         }
 
         internal static Exception ConvertTransferException(SocketException socketException, TimeSpan timeout, Exception originalException)
         {
             return ConvertTransferException(socketException, timeout, originalException,
-                false, null, TransferOperation.Undefined, null, TimeSpan.MaxValue);
+                TransferOperation.Undefined, false, null, TransferOperation.Undefined, null, TimeSpan.MaxValue);
         }
 
         private Exception ConvertObjectDisposedException(ObjectDisposedException originalException, TransferOperation transferOperation)
         {
             if (_timeoutErrorString != null)
             {
-                return ConvertTimeoutErrorException(originalException, _timeoutErrorString, _timeoutErrorTransferOperation);
+                return ConvertTimeoutErrorException(originalException, transferOperation, _timeoutErrorString, _timeoutErrorTransferOperation);
             }
             else if (_aborted)
             {
-                return new CommunicationObjectAbortedException(SR.SocketConnectionDisposed, originalException);
+                return new CommunicationObjectAbortedException(SR.Format(SR.SocketConnectionDisposed), originalException);
             }
```

Comment:
Remove SR.Format as unnecessary, just use the resource string directly

Reply:
👍